hd44780/gpio.go

```diff
@@ -0,0 +1,148 @@
+package hd44780
+
+import (
+	"errors"
+
+	"machine"
+)
+
+type GPIO struct {
+	dataPins []machine.GPIO
+	e        machine.GPIO
+	rw       machine.GPIO
+	rs       machine.GPIO
+
+	write func(data byte)
+	read  func() byte
+}
+
+func newGPIO(dataPins []uint8, e, rs, rw uint8, mode byte) Device {
+
+	pins := make([]machine.GPIO, len(dataPins))
+	for i := 0; i < len(dataPins); i++ {
+		m := machine.GPIO{Pin: dataPins[i]}
+		m.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+		pins[i] = m
+	}
+	enable := machine.GPIO{e}
+	enable.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+	registerSelect := machine.GPIO{rs}
+	registerSelect.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+	readWrite := machine.GPIO{rw}
+	readWrite.Configure(machine.GPIOConfig{Mode: machine.GPIO_OUTPUT})
+	readWrite.Low()
+
+	gpio := GPIO{
+		dataPins: pins,
+		e:        enable,
+		rs:       registerSelect,
+		rw:       readWrite,
+	}
+
+	if mode == DATA_LENGTH_4BIT {
+		gpio.write = gpio.write4BitMode
+		gpio.read = gpio.read4BitMode
+	} else {
+		gpio.write = gpio.write8BitMode
+		gpio.read = gpio.read8BitMode
+	}
+
+	return Device{
+		bus:        &gpio,
+		datalength: mode,
+	}
+}
+
+// SetCommandMode sets command/instruction mode
+func (g *GPIO) SetCommandMode(set bool) {
+	if set {
+		g.rs.Low()
+	} else {
+		g.rs.High()
+	}
+}
+
+func (g *GPIO) Write(data []byte) (n int, err error) {
```

Comment:
if it's exported, it should be docummented, Read function on this file too

Reply:
Done, although I think I should make GPIO unexported, it should not be used directly